### ISSUE TITLE
`mt.unique()` fails when passing an empty list as input

### ISSUE THREAD

**Issue by shizhiqiang007** [mars-project/mars#2059]
import mars.tensor as mt
mt.unique([]).execute(）

**Comment by qinxuye** [comment]
This error could only emerge for numpy 1.18.x, not for 1.19.x.

```
In [1]: import mars.tensor as mt

In [2]: mt.unique([]).execute()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-04232e429360> in <module>
----> 1 mt.unique([]).execute()

~/Workspace/mars/mars/core.py in execute(self, session, **kw)
    656 
    657         if wait:
--> 658             return run()
    659         else:
    660             thread_executor = ThreadPoolExecutor(1)

~/Workspace/mars/mars/core.py in run()
    652 
    653         def run():
--> 654             self.data.execute(session, **kw)
    655             return self
    656 

~/Workspace/mars/mars/core.py in execute(self, session, **kw)
    389 
    390         if wait:
--> 391             return run()
    392         else:
    393             # leverage ThreadPoolExecutor to submit task,

~/Workspace/mars/mars/core.py in run()
    384         def run():
    385             # no more fetch, thus just fire run
--> 386             session.run(self, **kw)
    387             # return Tileable or ExecutableTuple itself
    388             return self

~/Workspace/mars/mars/session.py in run(self, *tileables, **kw)
    504         tileables = tuple(mt.tensor(t) if not isinstance(t, (Entity, Base)) else t
    505                           for t in tileables)
--> 506         result = self._sess.run(*tileables, **kw)
    507 
    508         for t in tileables:

~/Workspace/mars/mars/session.py in run(self, *tileables, **kw)
    109             # set number of running cores
    110             self.context.set_ncores(kw['n_parallel'])
--> 111             res = self._executor.execute_tileables(tileables, **kw)
    112             return res
    113 

~/Workspace/mars/mars/utils.py in _inner(*args, **kwargs)
    458         def _inner(*args, **kwargs):
    459             with self:
--> 460                 return func(*args, **kwargs)
    461 
    462         return _inner

~/Workspace/mars/mars/executor.py in execute_tileables(self, tileables, fetch, n_parallel, n_thread, print_progress, mock, compose, name)
    881                     chunk_graph.compose(list(temp_result_keys))
    882                 # execute chunk graph
--> 883                 self.execute_graph(chunk_graph, list(temp_result_keys),
    884                                    n_parallel=n_parallel or n_thread,
    885                                    print_progress=print_progress, mock=mock,

~/Workspace/mars/mars/executor.py in execute_graph(self, graph, keys, n_parallel, print_progress, mock, no_intermediate, compose, retval, chunk_result)
    696             print_progress=print_progress, mock=mock, mock_max_memory=self._mock_max_memory,
    697             fetch_keys=fetch_keys, no_intermediate=no_intermediate)
--> 698         res = graph_execution.execute(retval)
    699         self._mock_max_memory = max(self._mock_max_memory, graph_execution._mock_max_memory)
    700         if mock:

~/Workspace/mars/mars/executor.py in execute(self, retval)
    577         # wait until all the futures completed
    578         for future in executed_futures:
--> 579             future.result()
    580 
    581         if retval:

~/miniconda3/envs/mars3.8/lib/python3.8/concurrent/futures/_base.py in result(self, timeout)
    437                 raise CancelledError()
    438             elif self._state == FINISHED:
--> 439                 return self.__get_result()
    440             else:
    441                 raise TimeoutError()

~/miniconda3/envs/mars3.8/lib/python3.8/concurrent/futures/_base.py in __get_result(self)
    386     def __get_result(self):
    387         if self._exception:
--> 388             raise self._exception
    389         else:
    390             return self._result

~/miniconda3/envs/mars3.8/lib/python3.8/concurrent/futures/thread.py in run(self)
     55 
     56         try:
---> 57             result = self.fn(*self.args, **self.kwargs)
     58         except BaseException as exc:
     59             self.future.set_exception(exc)

~/Workspace/mars/mars/utils.py in _inner(*args, **kwargs)
    458         def _inner(*args, **kwargs):
    459             with self:
--> 460                 return func(*args, **kwargs)
    461 
    462         return _inner

~/Workspace/mars/mars/executor.py in _execute_operand(self, op)
    444             # so we pass the first operand's first output to Executor.handle
    445             first_op = ops[0]
--> 446             self.handle_op(first_op, results, self._mock)
    447 
    448             # update maximal memory usage during execution

~/Workspace/mars/mars/executor.py in handle_op(self, *args, **kw)
    376 
    377     def handle_op(self, *args, **kw):
--> 378         return Executor.handle(*args, **kw)
    379 
    380     def _order_starts(self):

~/Workspace/mars/mars/executor.py in handle(cls, op, results, mock)
    647                 # The `UFuncTypeError` was introduced by numpy#12593 since v1.17.0.
    648                 try:
--> 649                     return runner(results, op)
    650                 except UFuncTypeError as e:
    651                     raise TypeError(str(e)).with_traceback(sys.exc_info()[2]) from None

~/Workspace/mars/mars/tensor/base/unique.py in execute(cls, ctx, op)
    405                     # axis cannot pass when dtype is object
    406                     kw['axis'] = op.axis
--> 407                 results = xp.unique(ar, **kw)
    408                 outs = op.outputs
    409                 if len(outs) == 1:

<__array_function__ internals> in unique(*args, **kwargs)

~/miniconda3/envs/mars3.8/lib/python3.8/site-packages/numpy/lib/arraysetops.py in unique(ar, return_index, return_inverse, return_counts, axis)
    273     # Must reshape to a contiguous 2D array for this to work...
    274     orig_shape, orig_dtype = ar.shape, ar.dtype
--> 275     ar = ar.reshape(orig_shape[0], -1)
    276     ar = np.ascontiguousarray(ar)
    277     dtype = [('f{i}'.format(i=i), ar.dtype) for i in range(ar.shape[1])]

ValueError: cannot reshape array of size 0 into shape (0,newaxis)
```

**Comment by wjsi** [comment]
Already fixed in #2049
